### PR TITLE
Alexey/adventure

### PR DESCRIPTION
**Pull request by Alexey-Slyusar**
The Logo Adventure Game written by Don Hopkins.A LLOGO port. It is ready to go into main branch, I think. The code illustrates LLOGO list processing capabilities and may be interesting for logo enthusiasts.

### CODE REVIEW

**Comment by Alexey-Slyusar** [comment]
The first file is old version that I mistakenly putted in wrong place. I'll delete the file.

**Comment by larsbrinkhoff** [comment]
I'd like a few changes made:

1. Rebase it to the current master, and resolve the conflict.  Probably just that the makefile SRC has changed.
2. Squash all commits together.
3. Rename ADV.LOGO to lowercase.

**Comment by Alexey-Slyusar** [comment]
>Rename ADV.LOGO to lowercase.
>I agree with Lars, please lowercase the ADV.LOGO filename in src/a2deh/

Done. Thanks.

**Comment by Alexey-Slyusar** [comment]
>Rebase it to the current master, and resolve the conflict. Probably just that the makefile SRC has >changed.
>Squash all commits together.

Lars, could you please suggest how can I do this properly.

File: ADV.LOGO [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+50 IF :RNUM = 4 MAKE "RNUM" 5 ELSE MAKE "RNUM" 4
+60 LOOK
+END
+
+TO A.FIX :ITEM					;FIX originally. conflicts with built-in procedrue
+10 IF IHAVE? :ITEM PR [YOU HAVE TO DROP IT TO FIX IT] CMD
+20 IF NOT HERE? :ITEM SEENO :ITEM
+30 IF NOT :ITEM = "MACHINE" PR [YOU CANT FIX THAT] CMD
+40 IF NOT ITEMLOC "WAND" = 0 PR [THE MACHINE IS NOT BROKEN] CMD
+50 IF NOT IHAVE? "SCREWDRIVER" PR [YOU DONT HAVE THE PRPPER TOOLS TO] PR [FIX IT] CMD
```
Comment (SimHacker) [inline ADV.LOGO]: Fix type "PRPPER" to more descriptive and better spelled "NECESSARY".

File: ADV.LOGO [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+60 PR [YOU FIX THE MACHINE WITH YOUR TRUSTY]
+70 PR [SCREWDRIVER. UPON BEING FIXED THE]
+80 PR [MACHINE STARTS UP AND PRODUCES A WAND]
```
Comment (SimHacker) [inline ADV.LOGO]: This is very exciting, so end the sentence with an !

File: ADV.LOGO [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+90 PUTITEM "WAND" 4
+100 CMD
+END
+
+TO A.EXAMINE :ITEM
+10 IF NOT HERE? :ITEM  PR [I SEE NO] :ITEM [HERE]  CMD
+20 IF :ITEM = "WAND" PR [IT BEARS A FADED INSCRIPTION:] PR [WAVE ME AND YOU'LL BE GLAD.] CMD
+30 IF NOT :ITEM = "MACHINE" PR SE [I SEE NOTHING SPECIAL ABOUT THE] PERIOD :ITEM CMD
+40 IF NOT 0 = ITEMLOC "WAND" PR [IT SEEMS TO BEAR THE MARKS OF A HASTY] PR [REPAIR JOB.] CMD
+50 PR [IT IS BROKEN YOU COULD FIX IT WITH]
```
Comment (SimHacker) [inline ADV.LOGO]: Reword "IT IS BROKEN BUT YOU COULD FIX IT WITH"

**Comment by Alexey-Slyusar** [comment]
> Probably just that the makefile SRC has changed.

Yes, that's right. I added "a2deh"  into makefile SRC section.

File: ADV.LOGO [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+TO INVENT
+10 PITEMS (- 1)
+20 CMD
+END
+
+TO HELP
+100 PR [TO MOVE TYPE]
+110 PR [NORTH EAST SOUTH WEST]
+120 PR [FOR NORTH SOUTH EAST WEST]
```
Comment (SimHacker) [inline ADV.LOGO]: Words don't correspond, not clear, not grammatically correct, and no punctuation. Reword: "TYPE NORTH, EAST, SOUTH OR WEST TO MOVE IN ANY DIRECTION."

File: ADV.LOGO [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+130 CR
+140 PR [TYPE LOOK TO SEE WHAT ROOM YOU]
+150 PR [ARE IN. YOU CAN TAKE AND DROP ITEMS]
```
Comment (SimHacker) [inline ADV.LOGO]: End sentence with period.

File: ADV.LOGO [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+160 PR [INVENT SHOWS YOUR INVENTORY]
```
Comment (SimHacker) [inline ADV.LOGO]: End sentence with period.

Comment (SimHacker) [reply ADV.LOGO]: Actually it would be better to phrase it as a call to action:
TYPE INVENT TO SHOW YOUR INVENTORY.

File: ADV.LOGO [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+170 PR [THE WORD IT MEANS THE LAST THING YOU]
+180 PR [REFERRED TO]
```
Comment (SimHacker) [inline ADV.LOGO]: End sentence in period.

File: ADV.LOGO [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+190 CR
+200 PR [THERE ARE SOME SPECIAL THINGS YOU CAN]
+210 PR [DO LIKE SAYING A.EXAMINE SOMETHING.]
+220 CR
+230 PR [TYPE SCORE TO SEE YOUR SCORE AND]
+240 PR [DONE TO QUIT.]
+250 PR [GOOD LUCK]                              ; PR [GOOD LUCK!] but LLOGO can't handle ! inside list.
```
Comment (SimHacker) [inline ADV.LOGO]: Aaaah I see. Rats! Well ignore my previous comment about using ! ... But at least end the sentence in a period, please.

**Comment by larsbrinkhoff** [comment]
> could you please suggest how can I do this properly.

Sure.  It's hard to say exactly, because I don't know which git you are using.

I would do it like with with command line git on LInux:

1. `git fetch` to update from origin repository.
2. `git rebase -i origin/master`
3. Now your editor should pop up with a file.
4. There should be many lines starting with `pick`.  Change all `pick` *but the first* to `f`.
5. Save the fle and exit the editor.
6. Now git will rebase all commits, and squash them in the same process.
7. But there will be a conflict, so git will stop and complain.
8. Edit Makefile to fix SRC.
9. `git add Makefile`
10. `git rebase --continue`
11. Hopefully, you should be done now.  Look at the result, is it good?  `git show` is fine?
12. `git push -f` to update the branch and this pull request.

File: ADV.LOGO (hunk 8 above)
Comment (Alexey-Slyusar) [reply ADV.LOGO]: @SimHacker, thank you! You commented old version file. Please take a look the file in  `src/a2deh/adv.logo` file. All punctuation preserved as in original version in that file. PRPPER is fixed too.

**Comment by Alexey-Slyusar** [comment]
@SimHacker, please review only  `src/a2deh/adv.logo` file. It preserves all original game's messages punctuation with fixes that you suggested to do (I meant PRPPER etc.).
I made some mess with the versions here in the branch so you commented an old version. Sorry for that.
And thank you for taking the time to look at the code.

**Comment by larsbrinkhoff** [comment]
> All punctuation preserved as in original version in that file.

We can have the cake and eat it too.  Make it two commits.  The first with the original version, and the second with fixed typos and punctuation.

**Comment by Alexey-Slyusar** [comment]
@larsbrinkhoff, thank you for the how to. I am working on a linux machine but I've never being using git command line interface, though. I use web interface instead. So it seems that it is time to learn a command line approach to work with git. Thank you!
But it takes some time in my case.

**Comment by Alexey-Slyusar** [comment]
>Now your editor should pop up with a file.
@larsbrinkhoff It seems that nano editor poped up or something like that. I prefer to use emacs for text editing. Is there a config file where I can change the default editor for git?

**Comment by Alexey-Slyusar** [comment]
Found this for default git editor change:
`config --global core.editor "emacs"`

**Comment by larsbrinkhoff** [comment]
That should work.  I'll usually just set the EDITOR environment variable.

**Comment by Alexey-Slyusar** [comment]
>I would do it like this with command line git on Linux:

I did the steps and it seems that all went fine. Thank you, Lars!
But I feel that I did it unconsciously, though...

**Comment by larsbrinkhoff** [comment]
I sometimes feel I should write a "Lars' guide to git rebasing".

**Comment by Alexey-Slyusar** [comment]
>I can't comment on the Logo code.
But you can play the game as Eric and my daughter already did :-)
Just type `?RF ADV LOGO DSK A2DEH` from `LLOGO` prompt on ITS.

In fact, I proud that such Zork veteran as Eric played the Logo Adventure and responded positively :-)

**Comment by larsbrinkhoff** [comment]
Testing the system by playing games is hard work, but someone's gotta do it!

**Comment by Alexey-Slyusar** [comment]
>I sometimes feel I should write a "Lars' guide to git rebasing".

Sorry for bothered you with that. 

I believe that I should learn git command line capabilities in much more depth for my own. 
But I believe also, that there will be one more newbie other than me here and the question arises again :-)

**Comment by eswenson1** [comment]
I’m with you Alexey. I’ve had to have Lars rescue me from Git hell several times. He’s a git wizard.

**Comment by eswenson1** [comment]
But Alexey, now you have to play Zork all the way through. And I guarantee, when you get as many points as you can, you will know it!

**Comment by Alexey-Slyusar** [comment]
>But Alexey, now you have to play Zork all the way through. 

Definitely!

>And I guarantee, when you get as many points as you can, you will know it!

:-)

**Comment by larsbrinkhoff** [comment]
This is good to go.  The convention we usually follow is that the submitter can click the "Rebase and merge" button.

**Comment by Alexey-Slyusar** [comment]
Thank you, Lars! I will follow the convention. I am just waiting for @SimHacker approval.

**Comment by SimHacker** [comment]
That last point is a real pain in the wazoo:

```
    +--v----v----v----v----v--+
    |         _______         |
    >  One   /       \     G  <
    | Lousy /         \    U  |
    > Point |   ___   |    E  <
    |       |  (___)  |       |
    >       <--)___(-->    P  <
    |       / /     \ \    o  |
    >      / /       \ \   s  <
    |     |-|---------|-|  t  |
    >     | |  \ _ /  | |  a  <
    |     | | --(_)-- | |  g  |
    >     | |  /| |\  | |  e  <
    |     |-|---|_|---|-|     |
    >      \ \__/_\__/ /      <
    |       _/_______\_       |
    >      |  f.m.i.c. |      <
    |      -------------      |
    >                         <
    |   Donald Woods, Editor  |
    >     Spelunker Today     <
    |                         |
    +--^----^----^----^----^--+

```

**Comment by SimHacker** [comment]
Ok I'm happy, is there anywhere I should click or is posting the Don Woods Stamp enough? ;)

**Comment by eswenson1** [comment]
@SimHacker Are you sure you're referring to the right adventure here?  The logo adv doesn't have this stamp.....

**Comment by larsbrinkhoff** [comment]
Someone brought up Zork in the comments.

**Comment by eswenson1** [comment]
> Someone brought up Zork in the comments.

Ah! I guess I was that someone!

**Comment by Alexey-Slyusar** [comment]
@SimHacker wrote:
>Ok I'm happy, is there anywhere I should click or is posting the Don Woods Stamp enough? ;)

I interpret it as approval, so ADV LOGO goes to the main branch now.

Many thanks to all here. It was fun.

**Pull request merged**